packages/apollo/src/commands/client/extract.ts

```diff
@@ -40,52 +24,31 @@ export default class ClientExtract extends ClientCommand {
   ];
 
   async run() {
-    const { clientIdentity, operations, filename }: any = await this.runTasks(
-      ({ flags, project, config, args }) => [
-        {
-          title: "Extracting operations from project",
-          task: async ctx => {
-            const operations = Object.values(
-              this.project.mergedOperationsAndFragmentsForService
-            ).map(operationAST => {
-              // While this could include dropping unused definitions, they are
-              // kept because the registered operations should mirror those in the
-              // client bundle minus any PII which lives within string literals.
-              const printed = printWithReducedWhitespace(
-                sortAST(hideCertainLiterals(operationAST))
-              );
-
-              return {
-                signature: manifestOperationHash(printed),
-                document: printed,
-                metadata: {
-                  engineSignature: engineSignature(operationAST)
-                }
-              };
-            });
-
-            ctx.operations = operations;
-            ctx.clientIdentity = config.client;
-          }
-        },
-        {
-          title: "Outputing extracted queries",
-          task: (ctx, task) => {
-            const filename = args.output;
-            task.title = "Outputing extracted queries to " + filename;
-            ctx.filename = filename;
-            writeFileSync(
-              filename,
-              JSON.stringify(
-                { version: 1, operations: ctx.operations },
-                null,
-                2
-              )
-            );
-          }
+    const { clientIdentity, operations, filename } = await this.runTasks<{
```

Comment:
I didn't realize that the `Result` from the `runTask` could be typed via these type variables!  Cool! (And huge win for this project!)

Reply:
For sure, I was pretty happy when I tried it on a whim and it was exactly as I expected. Intuitive APIs 🎉